### PR TITLE
Fix UAttributes integer type

### PR DESCRIPTION
**Pull request by stevenhartley**
There are a number of integer values that do not need to be signed but was set at int32 in lieu of uint32 by mistake, this change addresses the discrepancy. We also elaborate on what is means when the attribute is not present.

#106

### CODE REVIEW

File: uprotocol/uattributes.proto [hunk 1]
```diff
@@ -60,19 +60,20 @@ message UAttributes {
...
+    // TTL is How long this message should live for after it was generated (in milliseconds). If not present,
+    // the message is assumed to not timeout (should live forever)
     // Event expires when:
     // \$t_current > t_{id} + ttl\$
-    optional int32 ttl = 6;
+    optional uint32 ttl = 6;
     
     // Permission level per https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/basics/permissions.adoc[Permissions]
-    optional int32 permission_level = 7;
+    optional uint32 permission_level = 7;
```
Comment (PLeVasseur) [inline uprotocol/uattributes.proto:70]: Reading [thru the spec](https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/permissions.adoc), I see that the default when missing is set to 5 if an Android app and if not, it appears that's zero:
> For Android we will use the AOSP-standard manifest extension mechanism of <meta-data> to describe the permission level. We’ll use the name of uprotocol.permissionLevel and the value of the actual level, 0 to 5. The default value of 5 will be assigned implicitly when uprotocol.entity.permissionLevel` is not declared. 

> uPL value MUST be an integer, 0 to N, where N is the most permissive level. The default value is 0.

Is that why a default is not outlined here if not present? It's kind of implementation / host device dependent?

File: uprotocol/uattributes.proto [hunk 2]
```diff
@@ -60,19 +60,20 @@ message UAttributes {
     // https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/basics/qos.adoc[uProtocol Prioritization]
     UPriority priority = 5;
 
-    // TTL is How long this message should live for after it was generated (in milliseconds).
+    // TTL is How long this message should live for after it was generated (in milliseconds). If not present,
+    // the message is assumed to not timeout (should live forever)
```
Comment (sophokles73) [inline uprotocol/uattributes.proto:64]: In the code I see checks for `> 0` as well, which seems to suggest that we could/should rephrase this to
> If not present or 0 (zero), the message is assumed to not time out, i.e. should live forever.

File: uprotocol/uattributes.proto [hunk 3]
```diff
@@ -60,19 +60,20 @@ message UAttributes {
...
     // Permission level per https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/basics/permissions.adoc[Permissions]
-    optional int32 permission_level = 7;
+    optional uint32 permission_level = 7;
 
     // Communication error attribute populated by uP-L2 dispatchers only when an error 
     // has occurred in the delivery of RPC request or response events. 
     // The contents of this attribute, if present, is the unsigned integer representation of
-    // UCode
-    optional int32 commstatus = 8;
+    // UCode. If the attribute is not present, there is no communication error
```
Comment (sophokles73) [inline uprotocol/uattributes.proto:75]: I find it a little irritating that we refer to UCode values here but the type of the property is uint32 instead of UCode. Is there a particular reason for this? If so, I guess we should provide it here ...

File: uprotocol/uattributes.proto (hunk 1 above)
Comment (stevenhartley) [reply uprotocol/uattributes.proto:70]: investigating now... Great catch thank you @PLeVasseur !

File: uprotocol/uattributes.proto (hunk 2 above)
Comment (stevenhartley) [reply uprotocol/uattributes.proto:64]: fixed thank you

File: uprotocol/uattributes.proto (hunk 3 above)
Comment (stevenhartley) [reply uprotocol/uattributes.proto:75]: This is legacy that I dragged from CloudEvent, I fixed this now (it will be breaking change for the client libraries).

File: uprotocol/uattributes.proto [hunk 4]
```diff
@@ -71,9 +72,8 @@ message UAttributes {
 
     // Communication error attribute populated by uP-L2 dispatchers only when an error 
     // has occurred in the delivery of RPC request or response events. 
-    // The contents of this attribute, if present, is the unsigned integer representation of
-    // UCode. If the attribute is not present, there is no communication error
-    optional uint32 commstatus = 8;
+    // If the attribute is not present, there is no communication error
+    optional UCode commstatus = 8;
```
Comment (sophokles73) [inline uprotocol/uattributes.proto:76]: FMPOV this should be changed according to the outcome of https://github.com/eclipse-uprotocol/up-core-api/issues/77

**Review by sophokles73 — APPROVED** [review]
LGTM

File: uprotocol/uattributes.proto (hunk 4 above)
Comment (stevenhartley) [reply uprotocol/uattributes.proto:76]: @sophokles73 are you saying the documentation would need to change but not the field name and purpose right? We might add additional UCodes but it shouldn't change the signature

File: uprotocol/uattributes.proto (hunk 1 above)
Comment (PLeVasseur) [reply uprotocol/uattributes.proto:71]: Was this resolved? I think @stevenhartley was looking into it.

File: uprotocol/uattributes.proto (hunk 4 above)
Comment (sophokles73) [reply uprotocol/uattributes.proto:76]: Well, *communication error* is quite a generic term. IMHO it could be used to indicate both a problem at the infrastructure or the application level. This is what I suggested in #77. The only thing that we need to do is to define the semantics of *commstatus* accordingly in the spec.
